### PR TITLE
enable depth peeling by default

### LINKED ISSUE #527: Proposal to enable depth peeling by default
#450 brought depth peeling to PyVista which was a major improvement and sometimes I am amazed at how much better it makes a visualization. I'm thinking we should enable it by default as most users probably want its effects but won't know to use it.

Here's something I've been working on lately where the differences are stark:

Screen Shot 2020-01-09 at 10 57 53 PM

Does anyone have objections, concerns, or thoughts in general?

### PR DESCRIPTION
**Pull request by banesullivan**
Resolve #527 by controlling depth peeling (#450) in the `rcParams`. Default is now enabled.

### CODE REVIEW

File: pyvista/plotting/plotting.py [hunk 1]
```diff
@@ -243,11 +244,12 @@ def enable_depth_peeling(self, number_of_peels=5, occlusion_ratio=0.1):
             If True, depth peeling is supported.
 
         """
+        if number_of_peels is None:
+            number_of_peels = rcParams["depth_peeling"]["number_of_peels"]
         depth_peeling_supported = check_depth_peeling(number_of_peels,
                                                       occlusion_ratio)
         if hasattr(self, 'ren_win') and depth_peeling_supported:
             self.ren_win.AlphaBitPlanesOn()
-            self.ren_win.SetMultiSamples(0)
```
Comment (banesullivan) [inline pyvista/plotting/plotting.py]: This was supposed to get removed in the review of #450

**Comment by banesullivan** [comment]
@GuillaumeFavelier, are you cool with this? I imagine it will only benefit `mne-python`

**Comment by GuillaumeFavelier** [comment]
Yes, I'm okay with it!

Depending on user feedback I can always change it or report in here :+1:

**Comment by GuillaumeFavelier** [comment]
For now I have to disable the depth peeling in MNE, I noticed some artifacts in a use case with overlapping surfaces:

Without depth peeling | With depth peeling
---------------------------|------------------------
2020-01-14_1920x1080 | 2020-01-14_1920x1080

I tried different parameters (increasing the number of peels and decreasing the occlusion ratio) but I got no improvement.

To reproduce on [mne-python/master](https://github.com/mne-tools/mne-python/tree/634e30df5fdc644684a15c43330a2abfdb5e800c):

```sh
$ MNE_3D_BACKEND=pyvista ipython --gui=qt
In [1]: run tutorials/source-modeling/plot_visualize_stc.py
```
I have to create a minimal working example using pyvista only to help isolate the issue but it's totally possible that there is something wrong in my code.

**Comment by banesullivan** [comment]
Perhaps messing with the coincident topology options would help with that? Some details in https://github.com/pyvista/pyvista/issues/301#issuecomment-517943134

@GuillaumeFavelier, can you save out those two surfaces as VTK files and post them?

**Pull request merged**